contracts/minter-imago/src/contract.rs

```diff
@@ -0,0 +1,743 @@
+            code_id: msg.sg721_code_id,
+            msg: to_binary(&Sg721InstantiateMsg {
+                name: msg.sg721_instantiate_msg.name,
+                symbol: msg.sg721_instantiate_msg.symbol,
+                minter: env.contract.address.to_string(),
+                finalizer: msg.sg721_instantiate_msg.finalizer,
+                code_uri: msg.sg721_instantiate_msg.code_uri.to_string(),
+                collection_info: msg.sg721_instantiate_msg.collection_info,
+            })?,
+            funds: info.funds,
+            admin: Some(info.sender.to_string()),
+            label: String::from("Fixed price minter"),
+        }
+            .into(),
+        id: INSTANTIATE_SG721_REPLY_ID,
+        gas_limit: None,
+        reply_on: ReplyOn::Success,
+    }];
+
+    Ok(Response::new()
+        .add_attribute("action", "instantiate")
+        .add_attribute("contract_name", CONTRACT_NAME)
+        .add_attribute("contract_version", CONTRACT_VERSION)
+        .add_attribute("sender", info.sender)
+        .add_submessages(sub_msgs))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Mint {} => execute_mint_sender(deps, env, info),
+        ExecuteMsg::UpdateStartTime(time) => execute_update_start_time(deps, env, info, time),
+        ExecuteMsg::UpdatePerAddressLimit { per_address_limit } => {
+            execute_update_per_address_limit(deps, env, info, per_address_limit)
+        }
+        ExecuteMsg::MintTo { recipient } => execute_mint_to(deps, env, info, recipient),
+        ExecuteMsg::MintFor {
+            token_id,
+            recipient,
+        } => execute_mint_for(deps, env, info, token_id, recipient),
+        ExecuteMsg::SetWhitelist { whitelist } => {
+            execute_set_whitelist(deps, env, info, &whitelist)
+        }
+        ExecuteMsg::Withdraw {} => execute_withdraw(deps, env, info),
+        ExecuteMsg::BurnRemaining {} => execute_burn_remaining(deps, env, info),
+    }
+}
+
+pub fn execute_withdraw(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    if config.admin != info.sender {
+        return Err(ContractError::Unauthorized(
+            "Sender is not an admin".to_owned(),
+        ));
+    };
+
+    // query balance from the contract
+    let balance = deps
+        .querier
+        .query_balance(env.contract.address, NATIVE_DENOM)?;
+    if balance.amount.is_zero() {
+        return Err(ContractError::ZeroBalance {});
+    }
+
+    // send contract balance to creator
+    let send_msg = CosmosMsg::Bank(BankMsg::Send {
+        to_address: info.sender.to_string(),
+        amount: vec![balance],
+    });
+
+    Ok(Response::default()
+        .add_attribute("action", "withdraw")
+        .add_message(send_msg))
+}
```

Comment:
@shanev: this was removed from upstream, perhaps not needed here too?

Reply:
Yeah this can go away.